api/src/client/ApiTaxFilingClient.test.ts

```diff
@@ -0,0 +1,237 @@
+  return {
+    ApiKey: `some-ApiKey-${randomInt()}`,
+    Errors:
+      statusCode == 400
+        ? [
+            {
+              Error: `some-error-content-${randomInt()}`,
+              Field: randomInt() % 2 === 0 ? "Taxpayer ID" : "Business Name",
+            },
+          ]
+        : [],
+    Notice: statusCode == 200 ? "You are signed up for the Tax Calendar Reminder Service." : null,
+    StatusCode: statusCode as 200 | 400 | 500,
+    ...overrides,
+  };
+};
+
+describe("ApiTaxFilingClient", () => {
+  let client: ApiTaxFilingClient;
+  let logger: LogWriterType;
+  let taxIdAndBusinessNameAndEmail: { businessName: string; taxId: string; email: string };
+  const config = { apiKey: "abcdef", baseUrl: "example.com/gov2go" };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    taxIdAndBusinessNameAndEmail = generateTaxIdAndBusinessNameAndEmail({});
+    logger = LogWriter("NavigatorWebService", "ApiLogs", "us-test-1");
+    client = apiTaxFilingClient(config, logger);
+  });
+
+  describe("lookup", () => {
+    it("posts to the endpoint with the api tax-filing lookup request object", async () => {
+      const stubResponse = generateApiTaxFilingLookupResponse({});
+      mockAxios.post.mockResolvedValue({ data: stubResponse });
+
+      const postBody: ApiTaxFilingLookupRequest = {
+        ApiKey: config.apiKey,
+        ServiceName: "Search",
+        Data: [
+          {
+            Name: "taxpayer-id",
+            Value: taxIdAndBusinessNameAndEmail.taxId,
+          },
+          {
+            Name: "tax-business-name",
+            Value: taxIdAndBusinessNameAndEmail.businessName,
+          },
+        ],
+      };
+
+      await client.lookup(taxIdAndBusinessNameAndEmail.taxId, taxIdAndBusinessNameAndEmail.businessName);
+      expect(mockAxios.post).toHaveBeenCalledWith(`${config.baseUrl}/lookup`, postBody);
+    });
+
+    it("returns taxFiling data on success", async () => {
+      const Results = generateTaxFilingResults(["test-id"], 1) ?? [];
+      const stubResponse = generateApiTaxFilingLookupResponse({ Results }, "SUCCESS");
+      mockAxios.post.mockResolvedValue({ data: stubResponse });
+      const response = await client.lookup(
+        taxIdAndBusinessNameAndEmail.taxId,
+        taxIdAndBusinessNameAndEmail.businessName
+      );
+      expect(response.state).toEqual("SUCCESS");
+      expect(response.filings).toEqual([
+        { identifier: "test-id", dueDate: dateToShortISO(Results[0].Values[0]) },
+      ]);
+    });
+
+    it("returns only taxFiling state on failure", async () => {
+      const stubResponse = generateApiTaxFilingLookupResponse({}, "FAILED");
+      mockAxios.post.mockRejectedValue({ response: { data: stubResponse, status: 400 } });
+      const response = await client.lookup(
+        taxIdAndBusinessNameAndEmail.taxId,
+        taxIdAndBusinessNameAndEmail.businessName
+      );
+      expect(response.state).toEqual("FAILED");
+      expect(response.filings).toEqual([]);
```

Comment:
is there a reason why not to expect the whole object? It's nice, when looking at tests as documentation, to see exactly what the response would look like on failure, instead of just asserting on individual elements. As in, what is the `identifier` on failure? Is it skipped here because it's undefined? Or does it have a value that's not shown here? These are the questions that my brain asks and that could be clear in the test by asserting against the whole object